File: transform.py
import sys
import re

file_path = 'd:/Bilkheyr/resources/views/frontend/expense/add_all_expenses.blade.php'

# read original from backup
with open(file_path + '.backup', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract styles
style_match = re.search(r'<style>(.*?)</style>', content, re.DOTALL)
styles = style_match.group(1) if style_match else ''
styles = re.sub(r'\* \{.*?\}', '', styles, flags=re.DOTALL)
styles = re.sub(r'body \{.*?\}', '', styles, flags=re.DOTALL)

# Extract Main Container until end of body
main_match = re.search(r'(<!-- Main Container -->.*?<!-- Bootstrap 5 JS -->)', content, re.DOTALL)
main_content = main_match.group(1) if main_match else ''
main_content = main_content.replace('<!-- Bootstrap 5 JS -->', '')

# Extract scripts
script_match = re.search(r'<!-- Bootstrap 5 JS -->(.*?)</body', content, re.DOTALL)
scripts = script_match.group(1) if script_match else ''

new_content = f"""@extends('admin.admin_master')

@push('css')
    <!-- Bootstrap 5 CSS -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
    <style>
{styles}    </style>
@endpush

@section('admin')
{main_content}
@endsection

@push('scripts')
    <!-- Bootstrap 5 JS -->
{scripts}
@endpush
"""

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(new_content)

File: transform.php
<?php

$file = 'resources/views/frontend/report/party_wise_profit_Loss_reports.blade.php';
$content = file_get_contents($file);

preg_match('/<style>(.*?)<\/style>/s', $content, $styleMatch);
$css = $styleMatch[1] ?? '';

// We need to keep only targeted classes to avoid ruining other pages.
$allowedCssNames = [
    'stat-card', 'stat-icon', 'stat-value', 'stat-label',
    'filter-icon', 'profit-positive', 'profit-negative', 'badge',
    'report-footer', 'footer-item', 'footer-label', 'footer-value',
];

$finalCss = "";

// Simple css extraction block by block
preg_match_all('/([^{]+)\s*\{([^}]+)\}/s', $css, $cssBlocks, PREG_SET_ORDER);
foreach ($cssBlocks as $block) {
    $selector = trim($block[1]);
    $rules = trim($block[2]);

    $keep = false;
    foreach ($allowedCssNames as $allowed) {
        if (strpos($selector, '.' . $allowed) !== false) {
            $keep = true;
            break;
        }
    }

    if ($keep) {
        $finalCss .= "$selector {\n    $rules\n}\n\n";
    }
}

// Add variables to final CSS since they are used
$finalCss = ":root {\n    --primary-color: #004161;\n    --secondary-color: #99CC33;\n    --primary-dark: #002d47;\n}\n\n" . $finalCss;

preg_match('/<main class="main-content">(.*?)<\/main>/s', $content, $mainMatch);
$mainContent = rtrim($mainMatch[1] ?? '');

$bladeContent = <<<EOD
@extends('admin.admin_master')

@push('css')
<style>
$finalCss
</style>
@endpush

@section('admin')
<div class="page-content">
$mainContent
</div>
@endsection

EOD;

file_put_contents('resources/views/frontend/report/party_wise_profit_Loss_reports.blade.php', $bladeContent);
echo "Transformation complete.\n";
